src/base/static/components/templates/map.js

```diff
@@ -0,0 +1,163 @@
+import React, { Component, createRef } from "react";
+import PropTypes from "prop-types";
+import { connect } from "react-redux";
+import styled from "react-emotion";
+
+import MainMap from "../organisms/main-map";
+import ContentPanel from "../organisms/content-panel";
+import AddPlaceButton from "../molecules/add-place-button";
+import MapCenterpoint from "../molecules/map-centerpoint";
+import LeftSidebar from "../organisms/left-sidebar";
+import RightSidebar from "../organisms/right-sidebar";
+import GeocodeAddressBar from "../organisms/geocode-address-bar";
+
+import {
+  layoutSelector,
+  uiVisibilitySelector,
+} from "../../state/ducks/ui";
+import {
+  hasAnonAbilitiesInAnyDataset,
+  datasetSlugsSelector,
+} from "../../state/ducks/datasets-config";
+import {
+  placeConfigPropType,
+  placeConfigSelector,
+} from "../../state/ducks/place-config";
+import { hasGroupAbilitiesInDatasets } from "../../state/ducks/user";
+import { isLeftSidebarExpandedSelector } from "../../state/ducks/left-sidebar";
+import { isRightSidebarEnabledSelector } from "../../state/ducks/right-sidebar-config";
+import {
+  geocodeAddressBarEnabledSelector,
+  mapConfigSelector,
+  mapConfigPropType,
+} from "../../state/ducks/map-config";
+
+import {
+  getMainContentAreaWidth,
+  getMainContentAreaHeight,
+} from "../../utils/layout-utils";
+
+const MapContainer = styled("div")(props => ({
+  position: "relative",
+  overflow: "hidden",
+  height: getMainContentAreaHeight({
+    isContentPanelVisible: props.isContentPanelVisible,
+    isGeocodeAddressBarEnabled: props.isGeocodeAddressBarEnabled,
+    layout: props.layout,
+  }),
+  width: getMainContentAreaWidth({
+    isContentPanelVisible: props.isContentPanelVisible,
+    isRightSidebarVisible: props.isRightSidebarVisible,
+    layout: props.layout,
+  }),
+}));
+
+const SpotlightMask = styled("div")({
+  pointerEvents: "none",
+  position: "absolute",
+  left: "calc(50% - 100px)",
+  top: "calc(50% - 100px)",
+  width: "200px",
+  height: "200px",
+  borderRadius: "50%",
+  boxShadow:
+    "0px 0px 0px 800px rgba(0, 0, 0, 0.4), inset 0px 0px 20px 30px rgba(0, 0, 0, 0.4)",
+  zIndex: 8,
+});
```

Comment:
:tada: 

Hopefully this fixes that elusive off-center bug :)

Reply:
Yes, my sincere hope is that as of this PR all off-center issues are a thing of the past. 🤞